lib/linguist/heuristics.yml

```diff
@@ -277,6 +277,11 @@ disambiguations:
   - language: NL
     pattern: '^(b|g)[0-9]+ '
   - language: NewLisp
+- extensions: ['.p']
+  rules:
+  - language: Gnuplot
+    pattern: '^s?plot\b'
```

Comment:
Could you provide a bit of context on this heuristic rule? Does it mean that all Gnuplot files should contain `splot` or `plot` and if they don't, they're invalid?

Reply:
Most of them have, at least, one plot command (`plot` for 2D or `splot` for 3D). I think are still valid if there is no plot command, but useless. I found no example for that: https://github.com/search?q=language%3AGnuplot+set+-plot+-splot&type=Code

My guess is that a Gnuplot file with `.p` extension (rare) and no plot command (very rare) is very unlikely, so it's better than current situation, where all Gnuplot with `.p` extensions are misidentified.

Reply:
I think there's an issue with your search query: https://github.com/search?q=language%3AGnuplot+set+NOT+plot+NOT+splot&type=Code.

Reply:
Ouch, yes. I will do another iteration to refine the rule.